crypto/src/zcash_format.rs

```diff
@@ -138,6 +197,18 @@ pub mod test {
         assert_eq!(parse_g("0xc00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000").unwrap(), G2Affine::zero());
         assert_eq!(parse_g("0x93e02b6052719f607dacd3a088274f65596bd0d09920b61ab5da61bbdc7f5049334cf11213945d57e5ac7d055d042b7e024aa2b2f08f0a91260805272dc51051c6e47ad4fa403b02b4510b647ae3d1770bac0326a805bbefd48056c8c121bdb8").unwrap(), G2Affine::prime_subgroup_generator());
     }
+
+    #[test]
+    fn test_write_g1() {
+        assert_eq!(write_g(&G1Affine::zero()), "0xc00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000");
+        assert_eq!(write_g(&G1Affine::prime_subgroup_generator()), "0x97f1d3a73197d7942695638c4fa9ac0fc3688c4f9774b905a14e3a3f171bac586c55e83ff97a1aeffb3af00adb22c6bb");
+    }
+
+    #[test]
+    fn test_write_g2() {
+        assert_eq!(write_g(&G2Affine::zero()), "0xc00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000");
+        assert_eq!(write_g(&G2Affine::prime_subgroup_generator()), "0x93e02b6052719f607dacd3a088274f65596bd0d09920b61ab5da61bbdc7f5049334cf11213945d57e5ac7d055d042b7e024aa2b2f08f0a91260805272dc51051c6e47ad4fa403b02b4510b647ae3d1770bac0326a805bbefd48056c8c121bdb8");
+    }
```

Comment:
I like to add a proptest (i.e. fuzz-test) that constructs an arbitrary curve point and verifies that serialize-deserialize is the identity.